src/import-data/protected-data-utils.ts

```diff
@@ -45,13 +46,23 @@ export const isProtectedPfbSource = (pfbUrl: URL): boolean => {
   });
 };
 
+/**
+ * Determine if a TDR snapshot is considered protected data.
+ * */
+const isProtectedSnapshotSource = (snapshot: Snapshot): boolean => {
+  return snapshot.source.some((source) => source.dataset.secureMonitoringEnabled);
+};
+
 /**
  * Determine whether an import source is considered protected.
  */
 export const isProtectedSource = (importRequest: ImportRequest): boolean => {
   switch (importRequest.type) {
     case 'pfb':
       return isProtectedPfbSource(importRequest.url);
+    case 'tdr-snapshot-export':
+    case 'tdr-snapshot-reference':
+      return isProtectedSnapshotSource(importRequest.snapshot);
```

Comment:
I think this is a great improvement. In the future, it would be nice to improve pfb checks too, by storing the snapshot ID in the pfb (when a TDR snapshot is the source) and then using that to verify the protected status instead of the URL.

I'm not sure exactly how it overlaps with this work, but I know that TDR/WSM are planning to use TPS for more fine grained enforcement (rather than a single boolean protected/not-protected). At that point the check would look more like "WSM: can this workspace accept this snapshot"? and it would use TPS to check all the required restrictions.

Reply:
> I'm not sure exactly how it overlaps with this work, but I know that TDR/WSM are planning to use TPS for more fine grained enforcement (rather than a single boolean protected/not-protected). At that point the check would look more like "WSM: can this workspace accept this snapshot"? and it would use TPS to check all the required restrictions.

The next step of this ticket is to update the code that determines if an Azure workspace is "protected" based on its policies. So it'll be more general ("WSM: can this workspace accept protected data?") than specific to the snapshot, at least for now.

Reply:
We'll also possibly want to add something similar for Catalog datasets (though I don't know if the Catalog includes any protected data).

Reply:
Catalog datasets can contain protected data. Any catalog datasets that can be imported via this page (currently) are backed by a workspace or snapshot, so if necessary their IDs can be passed to the import code for this validation step. The original intent was that both the export and the validation would be done in the catalog service (the UI would pass the source catalog datasets and the destination workspace IDs), but that is a plan for a future that hasn't arrived yet.

Reply:
👍 We could load whatever information is needed for validation in `getCatalogDatasetImportRequest` and then use that in various other functions (`isProtectedSource`, etc.)

https://github.com/DataBiosphere/terra-ui/blob/b931b40ec60cda52fd567c050323b69e0ce1ab69/src/import-data/useImportRequest.ts#L112-L118

> The original intent was that both the export and the validation would be done in the catalog service (the UI would pass the source catalog datasets and the destination workspace IDs), but that is a plan for a future that hasn't arrived yet.

It looks like that is how it works now. The ImportData component passes a dataset ID and workspace ID to a Catalog endpoint.

https://github.com/DataBiosphere/terra-ui/blob/b931b40ec60cda52fd567c050323b69e0ce1ab69/src/import-data/ImportData.ts#L147-L151

Reply:
> It looks like that is how it works now. The ImportData component passes a dataset ID and workspace ID to a Catalog endpoint.

That's true, for a workspace import it does work "correctly". The snapshot route requires the browser to orchestrate the work. The catalog API should have a way to preflight the operation so only valid workspaces are shown / enabled in the UI. Right now the only source of protected data are snapshots but that won't be true for long.

Reply:
> That's true, for workspace-to-workspace import it does work "correctly". The snapshot route requires the browser to orchestrate the work. In both cases the catalog API should have a way to preflight the operation so only valid workspaces are shown / enabled in the UI.

Ah, I see after looking at DataBrowserDetails.js.

> Although maybe workspace cloning is already doing the right thing, since it doesn't go through this page?

I don't know how workspace cloning handles protected data. I'd guess all clones of a protected workspace are also considered protected / have the same requirements/logging.